framework/quota/test/org/apache/cloudstack/quota/QuotaStatementTest.java

```diff
@@ -0,0 +1,248 @@
+        assertTrue("period[0].before(period[1])" + period[0].toString(), period[0].before(period[1]));
+        assertTrue("period[0].get(Calendar.DATE) == 1" + period[0].toString(), period[0].get(Calendar.DATE) == 1);
+        assertTrue("period[0].get(Calendar.MONTH) == Calendar.OCTOBER" + period[0].toString(), period[0].get(Calendar.MONTH) == Calendar.OCTOBER); //october
+
+        //get last day of the previous month
+        aCalendar = Calendar.getInstance();
+        aCalendar.set(Calendar.MONTH, Calendar.DECEMBER);
+        aCalendar.set(Calendar.DATE, aCalendar.getActualMaximum(Calendar.DAY_OF_MONTH) + 1);
+        assertTrue(" period[1].get(Calendar.DATE) == aCalendar.get(Calendar.DATE)" + period[1].toString(), period[1].get(Calendar.DATE) == aCalendar.get(Calendar.DATE));
+        assertTrue("period[1].get(Calendar.MONTH) == aCalendar.get(Calendar.MONTH)" + period[1].toString(), period[1].get(Calendar.MONTH) == aCalendar.get(Calendar.MONTH));
+
+    }
+
+    @Test
+    public void testStatementPeriodHALFYEARLY() {
+        Calendar date = Calendar.getInstance();
+        Calendar aCalendar = Calendar.getInstance();
+
+        //QUATERLY
+        date = Calendar.getInstance();
+        date.set(Calendar.MONTH, Calendar.JANUARY); // 1 Jan
+        date.set(Calendar.DATE, 1);
+        Calendar period[] = quotaStatement.statementTime(date, STATEMENT_PERIODS.HALFYEARLY);
+        assertTrue(period != null);
+        assertTrue("period[0].before(period[1])" + period[0].toString(), period[0].before(period[1]));
+        assertTrue("period[0].get(Calendar.DATE) == 1" + period[0].toString(), period[0].get(Calendar.DATE) == 1);
+        assertTrue("period[0].get(Calendar.MONTH) == Calendar.JULY" + period[0].toString(), period[0].get(Calendar.MONTH) == Calendar.JULY); //july
+
+        //get last day of the previous month
+        aCalendar = Calendar.getInstance();
+        aCalendar.set(Calendar.MONTH, Calendar.DECEMBER);
+        aCalendar.set(Calendar.DATE, aCalendar.getActualMaximum(Calendar.DAY_OF_MONTH) + 1);
+        assertTrue(" period[1].get(Calendar.DATE) == aCalendar.get(Calendar.DATE)" + period[1].toString(), period[1].get(Calendar.DATE) == aCalendar.get(Calendar.DATE));
+        assertTrue("period[1].get(Calendar.MONTH) == aCalendar.get(Calendar.MONTH)" + period[1].toString(), period[1].get(Calendar.MONTH) == aCalendar.get(Calendar.MONTH));
+
+    }
+
+    @Test
+    public void testStatementPeriodYEARLY() {
+        Calendar date = Calendar.getInstance();
+        Calendar aCalendar = Calendar.getInstance();
+
+        //QUATERLY
+        date = Calendar.getInstance();
+        date.set(Calendar.MONTH, Calendar.JANUARY); // 1 Jan
+        date.set(Calendar.DATE, 1);
+        Calendar period[] = quotaStatement.statementTime(date, STATEMENT_PERIODS.YEARLY);
+        assertTrue("period != null", period != null);
+        assertTrue("period[0].before(period[1])" + period[0].toString(), period[0].before(period[1]));
```

Comment:
Add an assertion that the length of ``period`` is 2 to avoid a confusing ``ArrayIndexOutofBoundsException``.